draft-ietf-webpush-protocol.xml

```diff
@@ -327,6 +338,18 @@ Host: push.example.net
         that these URIs are not disclosed to unauthorized recipients (see <xref
         target="authorization"/>).
       </t>
+      <t>
+        The push service MAY provide a URI for a subscription set resource in a
+        link relation of type "urn:ietf:params:push:set". If provided, the push
+        service supports subscription sets. If available, the user agent SHOULD

```

Comment:
Oh, the missing piece there was something we discussed and agreed in Prague: namely that the push service can redirect requests to a subscription so that it was able to return the set instead.

That way, I think that you get what you want.  This allows you to provide a set for every subscription if that is what you want, avoiding any need to store anything for a subscription... until we get to filtering options later, that is, but I can explain how to hack around that.